integration/backup.py

```diff
@@ -0,0 +1,561 @@
+        }
+    },
+    {
+        "id": "55374edb-17de-487f-86cb-9a9fbc30823f",
+        "properties": {
+            "contents": "paragraph 3"
+        }
+    },
+    {
+        "id": "124ff6aa-597f-44d0-8c13-62fbb1e66888",
+        "properties": {
+            "contents": "paragraph 4"
+        }
+    },
+    {
+        "id": "f787386e-7d1c-481f-b8c3-3dbfd8bbad85",
+        "properties": {
+            "contents": "paragraph 5"
+        }
+    }
+]
+
+articles = [
+    {
+        "id": "2fd68cbc-21ff-4e19-aaef-62531dade974",
+        "properties": {
+            "title": "article a",
+            "datePublished": datetime.datetime.now(datetime.timezone.utc).isoformat()
+        }
+    },
+    {
+        "id": "7ea3f7b8-65fd-4318-a842-ae9ba38ffdca",
+        "properties": {
+            "title": "article b",
+            "datePublished": datetime.datetime.now(datetime.timezone.utc).isoformat()
+        }
+    },
+    {
+        "id": "769a4280-4b85-4e67-b685-07796c49a764",
+        "properties": {
+            "title": "article c",
+            "datePublished": datetime.datetime.now(datetime.timezone.utc).isoformat()
+        }
+    },
+    {
+        "id": "97fcc234-fd16-4a40-82bb-d614e9bddf8b",
+        "properties": {
+            "title": "article d",
+            "datePublished": datetime.datetime.now(datetime.timezone.utc).isoformat()
+        }
+    },
+    {
+        "id": "3fa435d3-6ab2-489d-abed-c25ec526c9f4",
+        "properties": {
+            "title": "article e",
+            "datePublished": datetime.datetime.now(datetime.timezone.utc).isoformat()
+        }
+    }
+]
+
+def make_backup_id():
+    return str(round(time.time() * 1000))
+
+class TestBackups:
+    def __init__(self, client: weaviate.Client):
+        client.schema.delete_all()
+        self.client = client
+        self.backup_dir = "/tmp/backups"
+
+    def test(self):
+        try:
+            self._create_and_restore_backup_with_waiting()
+            self._create_and_restore_backup_without_waiting()
+            self._create_and_restore_1_of_2_classes()
+            self._fail_creating_backup_on_non_existing_backend()
+            self._fail_checking_create_status_on_non_existing_backend()
+            self._fail_restoring_backup_on_non_existing_backend()
+            self._fail_creating_backup_for_non_existing_class()
+            self._fail_restoring_backup_for_existing_class()
+            self._fail_creating_existing_backup()
+            self._fail_checking_create_status_for_non_existing_backup()
+            self._fail_restoring_non_existing_backup()
+            self._fail_checking_restore_status_for_non_existing_restore()
+            self._fail_creating_backup_for_both_include_and_exclude_classes()
+            self._fail_restoring_backup_for_both_include_and_exclude_classes()
+            self._log("done!")
+        except:
+            self._cleanup()
+            raise TestFailedException("TestBackups failed")
+
+    @testmethod
```

Comment:
python has build in testing frameworks:
- unittest (which is a bit simpler, but less powerful) 
- pytest (my choice and very powerful, but some things are a bit clunky). 

Both have their own tmpDir/startup/shutdown/testdiscovery functionality.

I think you can write much better tests using those, happy to help :)

Reply:
@dirkkul I agree that upgrading our integration test suite with a framework would be nice to have, but it is outside the scope of this PR

Reply:
alright, I'll try to migrate everything at some point